src/main/java/io/github/pylonmc/pylon/base/items/Elevator.java

```diff
@@ -0,0 +1,126 @@
+import io.github.pylonmc.pylon.core.item.PylonItem;
+import io.github.pylonmc.pylon.core.item.builder.ItemStackBuilder;
+import io.github.pylonmc.pylon.core.util.gui.unit.UnitFormat;
+import net.kyori.adventure.text.Component;
+import net.kyori.adventure.text.ComponentLike;
+import org.bukkit.Location;
+import org.bukkit.Material;
+import org.bukkit.NamespacedKey;
+import org.bukkit.Sound;
+import org.bukkit.block.Block;
+import org.bukkit.entity.Player;
+import org.bukkit.event.player.PlayerToggleSneakEvent;
+import org.bukkit.inventory.ItemStack;
+import org.bukkit.persistence.PersistentDataContainer;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static io.github.pylonmc.pylon.base.util.KeyUtils.pylonKey;
+
+
+public class Elevator extends PylonBlock implements PylonSneakableBlock, PylonJumpableBlock {
+
+    public static class Item extends PylonItem {
+
+        public Item(@NotNull ItemStack stack) {
+            super(stack);
+        }
+
+        @Override
+        public @NotNull Map<String, ComponentLike> getPlaceholders() {
+            return Map.of("elevator_range",  UnitFormat.BLOCKS.format(getRange(getKey())));
+        }
+    }
+
+    public static final NamespacedKey ELEVATOR_1_KEY = pylonKey("elevator_1");
+    public static final NamespacedKey ELEVATOR_2_KEY = pylonKey("elevator_2");
+    public static final NamespacedKey ELEVATOR_3_KEY = pylonKey("elevator_3");
+
+    public static final Material MATERIAL = Material.QUARTZ_SLAB;
+
+    public static final ItemStack ELEVATOR_1_STACK = ItemStackBuilder.pylonItem(MATERIAL, ELEVATOR_1_KEY).build();
+    public static final ItemStack ELEVATOR_2_STACK = ItemStackBuilder.pylonItem(MATERIAL, ELEVATOR_2_KEY).build();
+    public static final ItemStack ELEVATOR_3_STACK = ItemStackBuilder.pylonItem(MATERIAL, ELEVATOR_3_KEY).build();
+
+    public static final ElevatorSettings ELEVATOR_1_SETTINGS = ElevatorSettings.fromConfig(Settings.get(ELEVATOR_1_KEY));
+    public static final ElevatorSettings ELEVATOR_2_SETTINGS = ElevatorSettings.fromConfig(Settings.get(ELEVATOR_2_KEY));
+    public static final ElevatorSettings ELEVATOR_3_SETTINGS = ElevatorSettings.fromConfig(Settings.get(ELEVATOR_3_KEY));
+
+    @SuppressWarnings("unused")
+    public Elevator(@NotNull Block block, @NotNull BlockCreateContext context) {
+        super(block);
+    }
+
+    @SuppressWarnings("unused")
+    public Elevator(@NotNull Block block, @NotNull PersistentDataContainer pdc) {
+        super(block);
+    }
+
+    public static int getRange(@NotNull NamespacedKey key) {
+        if (key.equals(ELEVATOR_1_KEY))
+            return ELEVATOR_1_SETTINGS.range();
+        if (key.equals(ELEVATOR_2_KEY))
+            return ELEVATOR_2_SETTINGS.range();
+        if (key.equals(ELEVATOR_3_KEY))
+            return ELEVATOR_3_SETTINGS.range();
+        return ELEVATOR_1_SETTINGS.range();
```

Comment:
```suggestion
        return getSetting("range", Integer.class)
```
You don't even need all this equalling, it'd make adding more tiers easy as well